### ISSUE TITLE
Entites should be added to the scene AFTER their properties have set.

### ISSUE THREAD

**Issue by greenfox1505** [codecat/godot-tbloader#86]
When [this line](https://github.com/codecat/godot-tbloader/blob/84a891f6499439c079833b8cf1c8335e8fec0c29/src/builder.cpp#L170) is called, the scenes `func _ready()` begins. When instantiating a scene in any other Godot-y method, the properties are all set *before* `add_child(thisChild)` is called.

![image](https://github.com/codecat/godot-tbloader/assets/1642297/aa8fcb5f-c294-4cfc-b615-9ed683ab6f25)

In this example, the value starts as `defaultValue`, then it gets set to `CHANGED_BY_INSPECTOR`, then `_ready()` runs, then lastly the value gets set to `changedByTrenchBroom`. Usually, I would add a `await owner.ready`, but even that is set *after* the object is added.

```gdscript
extends Node3D

@export var testString:String = "defaultValue":
	set(value):
		print("set from %s to %s" % [testString,value])
		testString = value

func _ready():
	print("ready! testString is %s" % testString)
	await get_parent().ready
	print("parent ready! testString is %s" % testString)
	pass
```

```log
Building map res://levels/tutorial/tutorial.map
set from defaultValue to CHANGED_BY_INSPECTOR
ready! testString is CHANGED_BY_INSPECTOR
set from CHANGED_BY_INSPECTOR to changedByTrenchBroom
parent ready! testString is changedByTrenchBroom
```

I believe if you move line 170 `m_loader->add_child(instance);` to after line 220 right before the `return`, it should behave more like what I would expect a Godot object to work. But I haven't tested this.

**Comment by codecat** [comment]
Good point! I'd have to test this, because I want to make sure it actually properly loads all of the properties correctly.

**Comment by greenfox1505** [comment]
Yeah, obviously you wanna test it before merging it. I'd do it myself if I wasn't in the middle of a game jam.

**Comment by Skaruts** [comment]
Not sure it makes much of a difference, unless you need to do some post-processing on the node. But for that I think it would probably be more reliable if TBLoader called a specific function on the nodes during the creation stage:

Something like: 

```cpp
void Builder::build_entity(int idx, LMEntity& ent, LMEntityGeometry& geo, const String& classname) {
    for (int i = 0; i < arr.size(); i++) {
        // (...)

        if (resource_loader->exists(path, "PackedScene")) {
            // (...)

            if (instance->has_method("on_import_finished")) {    // <--- this
                instance->call("on_import_finished");
            }

            return;
        }
    }

    UtilityFunctions::printerr("Path to entity resource could not be resolved: ", classname);
}
```

That should be enough for some post-processing (some other things would require a full second pass). I'm using this in my own slightly modified TBLoader, on an experimental `point_prop` entity. In TB I give that entity a `model` index (a bit of advanced fgd magic there to show the models in TB), and then in Godot I use that index to place a packed scene of the respective 3D model. Very experimental, but it's working.

So yea, for that I need the node to be `ready` before I process the property. In current state this isn't possible, unless you `await` in `_ready`, and I think it makes no difference if its parented before or after the properties are set.
